frontend/js/src/settings/music-services/details/MusicServices.tsx

```diff
@@ -469,8 +486,23 @@ export default function MusicServices() {
                     }}
                     name="lastFMStartDatetime"
                     title="Date and time to start import at"
+                    readOnly={!lastFMEdit && permissions.lastfm === "import"}
                   />
                 </div>
+                <div style={{ flex: 0, alignSelf: "end" }}>
+                  <button
+                    disabled={permissions.lastfm !== "import"}
+                    type={lastFMEdit ? "button" : "submit"}
+                    className={
+                      permissions.lastfm !== "import"
+                        ? "btn-default"
+                        : (lastFMEdit && "btn-success") || "btn-warning"
+                    }
```

Comment:
Bootstrap expects that buttons should have the `btn` class as well as the styling classes (btn-success, btn-warning, etc.) for proper styling.
Since this line will become a bit messy, I propose to refactor this ternary operator into a separate constant `lastFMEditButtonClass` declared above, then use it like soe:

```suggestion
                    className={`btn ${lastFMEditButtonClass}`}
```

Reply:
Ok, thanks for the review. Will make the changes.

Reply:
There was a misunderstanding with the solution I proposed.

I was not suggesting to use react state for this lastFMEditButtonClass, but rather a constant value (`const`) since it is simply derived from the state of permissions.lastfm and lastFMEdit.

Basically it would be equivalent to what you had before (ternary operator in className) except it is refactored so the value is defined higher up in the code to make it easier to read.

Reply:
I misread your text. Thanks for the clarification. I'll fix it immediately.

Reply:
@MonkeyDo, I think the notification didn't go through since I just commented. Please review the changes whenever you are free. No hurries.